packages/esm-patient-immunizations-app/src/immunizations/immunizations-form.component.tsx

```diff
@@ -40,17 +51,44 @@ const ImmunizationsForm: React.FC<DefaultWorkspaceProps> = ({ patientUuid, close
     formChanged: false,
   };
   const [formState, setFormState] = useState(initialState);
-  const updateSingle = <T extends keyof ImmunizationFormState>(name: T, value: typeof formState[T]) =>
+  const updateSingle = useCallback(<T extends keyof ImmunizationFormState>(name: T, value: typeof formState[T]) => {
     setFormState((state) => ({ ...state, [name]: value }));
+  }, []);
+
+  const onChangeComboBox = useCallback(
+    (event) => {
+      if (event.selectedItem) {
+        updateSingle('vaccineUuid', event.selectedItem.uuid);
+      } else {
+        updateSingle('vaccineUuid', null);
+      }
+    },
+    [updateSingle],
+  );
 
   const { t } = useTranslation();
   const currentUser = useSession();
   const { currentVisit } = useVisit(patientUuid);
   const isTablet = useLayoutType() === 'tablet';
+  const { data, isLoading } = useImmunizationsConceptSet('custom:(display,answers:(uuid,display))');
+  const vaccines = data?.answers;
+  const [vaccineState, setVaccineState] = useState(vaccines || []);
+  useEffect(() => {
+    if (!isLoading && vaccines?.length) {
+      setVaccineState(vaccines);
+    }
+  }, [vaccines, isLoading]);
```

Comment:
Why are you creating state for a value coming from a hook. This is not the right way.